Spec/Auth.swift

```diff
@@ -1106,6 +1106,36 @@ class Auth : QuickSpec {
                         done()
                     }
                 }
+                
+                tokenParams.clientId = "newClientId"
+                tokenParams.ttl = 2000
+                tokenParams.capability = "{ \"test:*\":[\"test\"] }"
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
+                        expect(error).to(beNil())
+                        guard let tokenRequest = tokenRequest else {
+                            XCTFail("TokenDetails is nil"); done(); return
+                        }
+                        expect(tokenRequest.clientId).to(equal("newClientId"))
+                        expect(tokenRequest.ttl).to(equal(2000))
+                        expect(tokenRequest.capability).to(equal("{ \"test:*\":[\"test\"] }"))
+                        done()
+                    }
+                }
+                
+                tokenParams.clientId = nil
+                
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.createTokenRequest(tokenParams, options: authOptions) { tokenRequest, error in
```

Comment:
Got it, I'll try to find out a way to test without assertions based on he token request properties.